packages/amplication-git-service/src/providers/githubProvider.ts

```diff
@@ -0,0 +1,323 @@
+  private static async createFilesTree(
+    octokit: Octokit,
+    owner: string,
+    repo: string,
+    latestCommitSha: string,
+    files: { path: string; content: string }[]
+  ): Promise<string> {
+    return (
+      await octokit.rest.git.createTree({
+        owner,
+        repo,
+        tree: files.map(file => {
+          return {
+            ...file,
+            type: 'blob',
+            mode: '100644'
+          };
+        }),
+        base_tree: latestCommitSha
+      })
+    ).data.sha;
+  }
+
+  private static async createCommit(
+    octokit: Octokit,
+    owner: string,
+    repo: string,
+    message: string,
+    latestCommitSha: string,
+    fileTreeSha: string
+  ): Promise<string> {
+    return (
+      await octokit.rest.git.createCommit({
+        owner,
+        repo,
+        message,
+        tree: fileTreeSha,
+        parents: [latestCommitSha]
+      })
+    ).data.sha;
+  }
+
+  private static async updateRef(
+    octokit: Octokit,
+    owner: string,
+    repo: string,
+    branch: string,
+    commitSha: string
+  ): Promise<string> {
+    return (
+      await octokit.rest.git.updateRef({
+        owner,
+        repo,
+        ref: `heads/${branch}`,
+        sha: commitSha
+      })
+    ).data.url;
+  }
+
+  public async commit(
+    branch: string,
+    message: string,
+    files: { path: string; content: string }[],
+    headCommit: string
+  ): Promise<string> {
+    const fileTreeSha = await GithubProvider.createFilesTree(
+      this.octokit,
+      this.owner,
+      this.repo,
+      headCommit,
+      files
+    );
+    const commitSha = await GithubProvider.createCommit(
+      this.octokit,
+      this.owner,
+      this.repo,
+      message,
+      headCommit,
+      fileTreeSha
+    );
+    await GithubProvider.updateRef(
+      this.octokit,
+      this.owner,
+      this.repo,
+      branch,
+      commitSha
+    );
+    return commitSha;
+  }
+
+  public async createPullRequest(
+    title: string,
+    body: string,
+    branch: string,
+    base: string
+  ): Promise<PullRequestMeta> {
+    try {
+      const { html_url: url, number } = (
+        await this.octokit.rest.pulls.create({
+          owner: this.owner,
+          repo: this.repo,
+          title,
+          body,
+          base,
+          head: branch
+        })
+      ).data;
+      return {
+        url,
+        number
+      };
+    } catch (err) {
+      switch (err?.response?.data?.errors[0]?.message) {
+        case `No commits between ${base.toLowerCase()} and ${branch.toLowerCase()}`:
+          throw new Error(`There is no new commit in branch ${branch}`);
+        case `A pull request already exists for ${this.owner}:${branch}.`:
+          throw new Error('Branch already has pull request');
+        default:
+          throw err;
+      }
+    }
+  }
+
+  public async addPullRequestComment(
+    pullRequestNumber: number,
+    body: string
+  ): Promise<string> {
+    try {
+      const { html_url } = (
```

Comment:
I find this part to be very unreadable.

`const { something } = someFunction.data`

I prefer to have separate variables instead for better clarity and readability.